### PR TITLE
Explicitly enable strip chopping for large uncompressed TIFFs

### LINKED ISSUE #5370: Loading large TIFF fails with OSError: -9
### What did you do?

Load a large TIFF created previously with Pillow.

### What did you expect to happen?

Load without error.

### What actually happened?

        Traceback (most recent call last):
          File "large_tiff_bug.py", line 15, in <module>
            img.load()
          File "/Users/mlg/code/stocks-map/venv/stocks-map/lib/python3.8/site-packages/PIL/TiffImagePlugin.py", line 1088, in load
            return self._load_libtiff()
          File "/Users/mlg/code/stocks-map/venv/stocks-map/lib/python3.8/site-packages/PIL/TiffImagePlugin.py", line 1192, in _load_libtiff
            raise OSError(err)
        OSError: -9

### What are your OS, Python and Pillow versions?

* OS: macOS 11.2.3
* Python: 3.8.8
* Pillow: 8.1.2
* PIL.features.version_codec('libtiff'): 4.2.0

Code:
```python
from PIL import Image

width  = 30_000
height = 35_792

print(f'creating test image: {width:,} x {height:,} = {width * height:,}; {width * height * 2:,} bytes')
img = Image.new( 'I;16', (width, height) ) # I'm using 16 bit greyscale 
img.save( '_tmp.tiff', compression='tiff_adobe_deflate' ) # add zip compression, so filesize doesn't get out of hand

print()
print(f'loading test image')
Image.MAX_IMAGE_PIXELS = 1_800_000_000
img = Image.open('_tmp.tiff')
img.load()
```

I believe we are hitting the size limit limit [here](https://github.com/python-pillow/Pillow/blob/master/src/libImaging/TiffDecode.c#L489), because reducing `height` to `35_791` in the above code makes it work.

30.000 x 35.792 x 2 bytes = 2.147.520.000 bytes -> error -9 ([IMAGING_CODEC_MEMORY](https://github.com/python-pillow/Pillow/blob/master/src/libImaging/Imaging.h#L686))
30.000 x 35.791 x 2 bytes = 2.147.460.000 bytes  -> works

This leads me to believe that INT_MAX is 2.147.483.647, so it's an `int32` (32 bit, signed), where it could be 64 bit (on supported platforms) and unsigned to maximize the size of loadable images.

In addition I noticed the same image CAN be loaded by pillow when re-saved in Preview.app (or Affinity Photo for that matter). It looks like Preview.app saves it in strips of a single row each, which adds some metadata but gets around the memory limit. Pillow seems to save a single strip containing all rows. So it would be cool to be able to control layout of strips when saving tiffs, as it would allow to get around the memory problem as well.

Attachment: [Example TIFF + TIFF saved with Preview.app](https://github.com/python-pillow/Pillow/files/6237890/large_tiffs.zip)

### PR DESCRIPTION
**Pull request by kmilos**
Helps #5370 (only works for uncompressed images)

### CODE REVIEW

**Comment by radarhere** [comment]
Would you be able to add a test, that fails without your change and passes with it?

**Comment by kmilos** [comment]
Hm, [I just pushed a test to my branch](https://github.com/kmilos/Pillow/commit/1fc144089e8fdfed42df05e2ecc896280ab287df), but it's not showing up here?!

**Comment by kmilos** [comment]
Ah, only internal `TiffImagePlugin.ImageFileDirectory_v2` is ever used to parse the TIFF IFDs (in both `Image.open()` and `TiffImagePlugin._load_libtiff()`, there's no way to actually see how libtiff is re-interpreting the IFD as far as I can tell, so not really sure how to test this...

File: Tests/test_file_libtiff.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 import pytest
 
 from PIL import Image, ImageFilter, TiffImagePlugin, TiffTags, features
-from PIL.TiffImagePlugin import SUBIFD
+from PIL.TiffImagePlugin import SUBIFD, STRIPOFFSETS
```
Comment (radarhere) [inline Tests/test_file_libtiff.py:12]:
```suggestion
from PIL.TiffImagePlugin import STRIPOFFSETS, SUBIFD
```
Lint fix.

Comment (kmilos) [reply Tests/test_file_libtiff.py:12]: Saw that, but didn't bother since the test is not working...

**Comment by radarhere** [comment]
https://linux.die.net/man/3/tifffdopen
> The default behaviour, for backwards compatibility, is to enable strip chopping.

Is it possible that this PR doesn't actually do anything?

**Comment by kmilos** [comment]
I only know this: https://gitlab.com/libtiff/libtiff/-/blob/29219a5bd14040d60e376324a52ef91b68808143/libtiff/tif_open.c#L254-257

But it does seem the flags are initialized by default to do the chopping: https://gitlab.com/libtiff/libtiff/-/blob/29219a5bd14040d60e376324a52ef91b68808143/libtiff/tif_open.c#L161-164 

However, there was also recent (not yet released) PR that seems to indicate the default wasn't working on libtiff CMake builds (which is what Pillow uses lately AFAIK): https://gitlab.com/libtiff/libtiff/-/merge_requests/271

If this is the behaviour Pillow was expecting, doesn't hurt to be explicit ;)

**Comment by radarhere** [comment]
Ok, if the default has been wrong in previous libtiff versions, then it seems like this would be good, as it will fix that.

**Comment by radarhere** [comment]
libtiff 4.4.0 has now been released, fixing the default behaviour.

**Pull request merged**